src/jdk.compiler/share/classes/com/sun/tools/javac/parser/ReferenceParser.java

```diff
@@ -113,9 +111,12 @@ public Reference parse(String sig) throws ParseException {
 
         try {
             int slash = sig.indexOf("/");
+            int hash = sig.indexOf("#");
+            if (hash > -1 && slash > hash) {
+                // A slash following a hash is part of a doc-file path, not a module separator
+                slash = -1;
+            }
```

Comment:
I thought we were backing off file references ...

Reply:
Changed back to old code